### DISCUSSION TITLE
Idle systems stop logging

### DISCUSSION THREAD

**Discussion by vanquishkuso** [plausible/analytics#4081] [Self-Hosted Support] [answered]
Hello everyone,

We have some kiosk systems, a few run on Next.js and the others run on React at a total of around 40 systems. Recently we added tracking with Plausible to see what pages were used most, however, they all seem to drop connection to Plausible after some time. These systems are constant online and may stay on the same page for a long period of time. Does a prolonged idle time cause Plausible to close the connection? We have other projects that are like regular websites and the tracking works fine for those.

I know this is not the main use case for Plausible, is there any way to make this work? Thankful for any help!

![image](https://github.com/plausible/analytics/assets/46133585/f6bd8ad1-2662-44c5-80f1-48773df6da21)

**Comment by ruslandoga** [comment, marked as answer]
👋 @vanquishkuso 

> Does a prolonged idle time cause Plausible to close the connection?

Yes, [Cowboy,](https://github.com/ninenines/cowboy) the webserver used by Plausible, drops connections that are idle (no data sent from the client) after 60 seconds. The option is called [`idle_timeout`](https://github.com/search?q=repo%3Aninenines%2Fcowboy%20idle_timeout&type=code)

> [...] is there any way to make this work?

Starting in [v2.1.0-rc.1](https://github.com/plausible/analytics/discussions/4084) you would be able to mount [custom configs](https://github.com/plausible/analytics/pull/3906) into Plausible and configure that value like this

```elixir
import Config

# overrides https://github.com/plausible/analytics/blob/435d7082891004bc55e740c6810fad8f70a09b62/config/runtime.exs#L308-L313 by adding an infinite idle_timeout
config :plausible, PlausibleWeb.Endpoint,
  http: [
    port: 8000,
    ip: {0, 0, 0, 0},
    transport_options: [max_connections: :infinity],
    protocol_options: [max_request_line_length: 8192, max_header_value_length: 8192, idle_timeout: :infinity]
  ]
```

**Reply by ruslandoga** [reply]
v2.1.0-rc.1 has been published: https://github.com/plausible/analytics/discussions/4084

**Comment by vanquishkuso** [comment]
Thank you for the fast reply! You answered my other question already with your edit 🤩. 

Visit duration and so on is not really of interest for these systems anyway, but if an idle system gets interaction (e.g. route changes) does Plausible track these events or does it require the browser to refresh first?

**Reply by ruslandoga** [reply]
Browser refresh is not required. Regarding tracking, if the interactions happen within 30 minutes of each other, they would be considered a single session, if they are more than 30 minutes apart, they would be considered separate sessions. AFAIK that only affects visit durations (session end - session start), visits count (visits = sessions), and some other metrics session-level metrics that you say are not of interest :)

**Comment by ruslandoga** [comment]
I wonder though, why does it matter that Plausible closes the connection. Wouldn't the new event cause the browser to reconnect since it makes another XHR request?

**Reply by vanquishkuso** [reply]
Hopefully, I'll have to test and see, hopefully it gets the page events needed :).
